### PR TITLE
GH_903 AmCharts Value Field Fix

### PR DESCRIPTION
**Pull request by mlzummo**
Value Field Fix
Fixes GH-903

Signed-off-by: Mathew Zummo <mzummo@gmail.com>

### CODE REVIEW

File: src/amchart/CommonSerial.js [hunk 1]
```diff
@@ -488,15 +488,11 @@
...
-        var context = this;
         this._categoryField = this.columns()[0];
         this._chart.categoryField = this.columns()[0];
-        this._valueField = [];
         this._openField = [];
         this._closeField = [];
-        this.columns().slice(1, this.columns().length).forEach(function(col, idx) {
-            context._valueField.push(col);
-        });
+        this._valueField = _.filter(function (d, i) { return i > 0; });
```
Comment (GordonSmith) [inline src/amchart/CommonSerial.js]: The following would be better:
```
this._valueField = this.columns().slice(1);
```

Comment (GordonSmith) [inline src/amchart/CommonSerial.js]: "_" is undefined

**Comment by dtsnell4** [comment]
looks good

File: src/amchart/CommonSerial.js [hunk 2]
```diff
@@ -487,16 +487,12 @@
...
+    CommonSerial.prototype.amFormatColumns = function() {
         this._categoryField = this.columns()[0];
         this._chart.categoryField = this.columns()[0];
-        this._valueField = [];
         this._openField = [];
         this._closeField = [];
-        this.columns().slice(1, this.columns().length).forEach(function(col, idx) {
-            context._valueField.push(col);
-        });
+        this._valueField = this.columns().filter(function (d, i) { return i > 0; });
```
Comment (GordonSmith) [inline src/amchart/CommonSerial.js]: I think the original suggestion was that:
```
this._valueField = this.columns().slice(1);
```
Was better?

**Comment by mlzummo** [comment]
@GordonSmith Please re-review removed unnecessary code

**Comment by GordonSmith** [comment]
Needs rebase

**Comment by mlzummo** [comment]
@GordonSmith Rebased

**Comment by dtsnell4** [comment]
looks ok to me

File: demos/test-no-amd.html [hunk 3]
```diff
@@ -28,7 +28,7 @@
     <script src="../bower_components/dagre/index.js"></script>
     <script src="../bower_components/colorbrewer/colorbrewer.js"></script>
     <script src="../bower_components/topojson/topojson.js"></script>
-    <script src="../bower_components/d3-cloud/d3.layout.cloud.js"></script>
+    <script src="../bower_components/d3-cloud/build/d3.layout.cloud.js"></script>
```
Comment (GordonSmith) [inline demos/test-no-amd.html]: No longer needed

File: src/config.js [hunk 4]
```diff
@@ -14,7 +14,7 @@ require.config({
         "dagre": "../bower_components/dagre/index",
         "topojson": "../bower_components/topojson/topojson",
         "colorbrewer": "../bower_components/colorbrewer/colorbrewer",
-        "d3-cloud": "../bower_components/d3-cloud/d3.layout.cloud",
+        "d3-cloud": "../bower_components/d3-cloud/build/d3.layout.cloud",
```
Comment (GordonSmith) [inline src/config.js]: no longer needed

File: src/other/WordCloud.js [hunk 5]
```diff
@@ -5,7 +5,7 @@
     } else {
         root.other_WordCloud = factory(root.d3, root.common_SVGWidget, root.other_IWordCloud);
     }
-}(this, function (d3, SVGWidget, IWordCloud) {
+}(this, function (d3, SVGWidget, IWordCloud, d3Cloud) {
```
Comment (GordonSmith) [inline src/other/WordCloud.js]: no longer needed

File: src/other/WordCloud.js [hunk 6]
```diff
@@ -38,7 +38,12 @@
     };
 
     WordCloud.prototype.enter = function (domNode, element) {
-        this.cloud = d3.layout.cloud()
+        if (typeof define === "function" && define.amd) {
+            this.cloud = d3Cloud();
+        } else {
+            this.cloud = d3.layout.cloud();
+        }
+        this.cloud
```
Comment (GordonSmith) [inline src/other/WordCloud.js]: no longer needed

File: test/runner_build_nonamd.html [hunk 7]
```diff
@@ -28,7 +28,7 @@
     <script src="../bower_components/dagre/index.js"></script>
     <script src="../bower_components/colorbrewer/colorbrewer.js"></script>
     <script src="../bower_components/topojson/topojson.js"></script>
-    <script src="../bower_components/d3-cloud/d3.layout.cloud.js"></script>
+    <script src="../bower_components/d3-cloud/build/d3.layout.cloud.js"></script>
```
Comment (GordonSmith) [inline test/runner_build_nonamd.html]: no Longer needed.

**Pull request merged**